src/main/java/uk/gov/hmcts/reform/managecase/zuulfilters/UrlWhitelistRoutingFilter.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.hmcts.reform.managecase.zuulfilters;
+
+import com.netflix.zuul.ZuulFilter;
+import com.netflix.zuul.context.RequestContext;
+import com.netflix.zuul.exception.ZuulException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.netflix.zuul.util.ZuulRuntimeException;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.managecase.ApplicationParams;
+
+import java.util.List;
+
+import static org.springframework.cloud.netflix.zuul.filters.support.FilterConstants.PRE_TYPE;
+import static org.springframework.cloud.netflix.zuul.filters.support.FilterConstants.SIMPLE_HOST_ROUTING_FILTER_ORDER;
+
+@Component
+public class UrlWhitelistRoutingFilter extends ZuulFilter {
+
+    @Autowired
+    private ApplicationParams applicationParams;
+
+    @Override
+    public String filterType() {
+        return PRE_TYPE;
+    }
+
+    @Override
+    public int filterOrder() {
+        return SIMPLE_HOST_ROUTING_FILTER_ORDER - 2;
+    }
+
+    @Override
+    public boolean shouldFilter() {
+        return true;
+    }
+
+    @Override
+    public Object run() {
+        RequestContext context = RequestContext.getCurrentContext();
+
+        validateWhitelistedUrls(context);
+
+        return null;
+    }
+
+    private void validateWhitelistedUrls(RequestContext context) {
```

Comment:
what if you define multiple paths in the config rather than custom filter? 
`zuul:
  routes:
    search-cases:
      path: /ccdsearchCases/**
      url: data-store
    my_another_path:
      path: /anotherpath/**
      url: data-store`

Reply:
This is not going to work, as zuul drops all that is before ** There is a way to configure it not to drop prefix, but then we would have to do it somewhere manually. Additionally it will allow everything in the ** placeholder, so I think regex is more precise.

Reply:
Ok, i was thinking something like below. Don't worry if it's not working 😄 
zuul:
  routes:
    search-cases::
      path: /ccd/searchCases
      url: data-store
    internal-search-cases:
      path: /ccd/internal/searchCases
      url: data-store